lang/funcs/simple/env_func.go

```diff
@@ -0,0 +1,87 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// functions to look up variables in the mgmt server instance environment
+
+package simple // TODO: should this be in its own individual package?
+
+import (
+	"os"
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	Register("getenv", &types.FuncValue{
+		T: types.NewType("func(str) str"),
+		V: getenv,
+	})
+	Register("defaultenv", &types.FuncValue{
+		T: types.NewType("func(str, str) str"),
+		V: defaultenv,
+	})
+	Register("hasenv", &types.FuncValue{
+		T: types.NewType("func(str) bool"),
+		V: hasenv,
+	})
+	Register("env", &types.FuncValue{
+		T: types.NewType("func() {str: str}"),
+		V: env,
+	})
+}
+
+// getenv gets environment variable by name or empty string if non existing.
+func getenv(input []types.Value) (types.Value, error) {
+	value := os.Getenv(input[0].Str())
+	return &types.StrValue{
+		V: value,
+	}, nil
+}
+
+// defaultenv gets environment variable by name or return default if non existing.
```

Comment:
Make all the functions public. eg:

`Defaultenv` (capitalized)

so they'll appear in docs :)

Reply:
I didn't know the capitalisation mattered :) Go is weird sometimes. Consistent but weird.

Reply:
Yup. And it's capitalization of the actual function ;) The comment just has to match or golint will fail.